src/web/query_socket.cpp

```diff
@@ -63,6 +72,17 @@ query_socket::query_socket(zmq::context& context, server_node& node,
         request.enqueue(to_chunk(hash));
     };
 
+    const auto encode_hash_or_height = [&](zmq::message& request,
```

Comment:
Avoid the conditionality here, create independent handlers.

Reply:
The method calls the independent handlers, the issue is that `arguments` can contain either a hash or a height.  I agree that the decode isn't the safest way to determine which it is though.  A lexical cast for height's type should work (which should throw if given a hash as input).

Reply:
Do you mean that there is a single RPC method call (same name) on the wire that allows either parameter type?

Reply:
Yes, there is, but there doesn't have to be.  We can support some calls by either (e.g. getblock/getblockheader).  Not required, but I thought it would be nice :-)  I see why it's dangerous too though, so I'll separate these properly.

Reply:
`encode_hash_or_height` is not yet removed in this revision.  It's marginally safer (and can properly error now), but can still be removed.